jerry-core/vm/vm.c

```diff
@@ -1651,6 +1657,36 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_ADD:
         {
+          if (ecma_are_values_integer_numbers (left_value, right_value))
```

Comment:
That's reasonable, I agree that introducing macros here would not be beneficial to the maintainability of the code.  
Since you don't see any code size regressions I think it's fine to go ahead as is.